www/js/lib/uiUtil.js

```diff
@@ -1003,6 +1011,436 @@ function warnAndOpenExternalLinkInNewTab (event, clickedAnchor, archive) {
+        triangleX = event.clientX - divRectX - 20;
+    }
+    // If right edge of div is greater than margin from the right side of window, shift it to margin
+    if (divRectX + divWidth > screenWidth - margin) {
+        triangleX += divRectX;
+        divRectX = screenWidth - divWidth - margin;
+        triangleX -= divRectX;
+    }
+    // If we're less than margin to the left, shift it to margin px from left
+    if (divRectX < margin) {
+        triangleX += divRectX;
+        divRectX = margin;
+        triangleX -= divRectX;
+    }
+    // Adjust triangleX if necessary
+    if (triangleX < 10) triangleX = 10;
+    if (triangleX > divWidth - 10) triangleX = divWidth - 10;
+    // Now set the calculated x and y positions
+    div.style.top = divRectY + win.scrollY + 'px';
+    div.style.left = divRectX + 'px';
+    div.style.opacity = '1';
+    // Now create the arrow span element. Note that we cannot attach it yet as we need to populate the div first
+    // and doing so will overwrite the innerHTML of the div
+    const triangleColour = '#b7ddf2'; // Same as border colour of div
+    const span = document.createElement('span');
+    span.style.cssText = `
+        width: 0;
+        height: 0;
+        border-${triangleDirection}: 16px solid ${triangleColour};
+        border-left: 8px solid transparent !important;
+        border-right: 8px solid transparent !important;
+        position: absolute;
+        top: ${triangleY}px;
+        left: ${triangleX}px;
+    `;
+    return { div: div, span: span };
+}
+
+/**
+ * Adds event listeners to the popover's control icons
+ * @param {Element} anchor The anchor which launched the popover
+ * @param {Element} popover The containing element of the popover (div)
+ * @param {Document} doc The doucment on which to operate
+ */
+function addEventListenersToPopoverIcons (anchor, popover, doc) {
+    const breakout = function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        // Adding the newcontainer property to the anchor will be cauught by the filterClickEvent function and will open in new tab
+        anchor.newcontainer = true;
+        anchor.click();
+        closePopover(popover);
+    }
+    const closeIcon = doc.getElementById('popcloseicon');
+    const breakoutIcon = doc.getElementById('popbreakouticon');
+    // Register mousedown event (should work in all contexts)
+    closeIcon.addEventListener('mousedown', function () {
+        closePopover(popover);
+    }, true);
+    breakoutIcon.addEventListener('mousedown', breakout, true);
+    // Additionally register either pointerdown or touchstart if supported for faster response
+    const eventName = window.PointerEvent ? 'pointerdown' : 'touchstart';
+    closeIcon.addEventListener(eventName, function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        closePopover(popover);
+    }, true);
+    breakoutIcon.addEventListener(eventName, breakout, true);
```

Comment:
Code may not be necessary if `mousedown` is universally supported. Can't remember why I added it -- I think I was having trouble getting it to work at some point, but logically shouldn't be needed. Will test.